local.tf

```diff
@@ -16,7 +16,7 @@ locals {
   worker_group_launch_template_count = length(var.worker_groups_launch_template)
 
   worker_has_linux_ami   = length([for x in concat(var.worker_groups, var.worker_groups_launch_template) : x if lookup(x, "platform", "linux") == "linux"]) > 0
-  worker_has_windows_ami = length([for x in concat(var.worker_groups, var.worker_groups_launch_template) : x if lookup(x, "platform", "linux") == "windows"]) > 0
+  worker_has_windows_ami = length([for x in concat(var.worker_groups, var.worker_groups_launch_template) : x if lookup(x, "platform", "windows") == "windows"]) > 0
```

Comment:
I think if we do this, we'll always lookup for windows and linux AMI if someone doesn’t explicitly set `platform` on worker nodes. Because both `worker_has_linux_ami` and `worker_has_windows_ami` will be `true`.

I don't know how to solve this without introducing cycle. Maybe it worth to add new variables to explicitly ask for lookup for plateform specific AMI. Something like `worker_search_ami_linux` and `worker_search_ami_windows`. Both will be set to `true` by default.

Reply:
@barryib - ah right.

What about changing the default return on the look from `lookup(x, "platform", "windows") == "windows"`  to `lookup(x,"platform",local.workers_group_defaults["platform"])`  ?

Or instead of using the concat + lookup, maybe the `worker_has_windows_ami` could be simplified to `worker_has_windows_ami = local.workers_group_defaults["platform"] == "windows"`? I haven't looked through the module enough to understand the logic yet, i may be overlooking the module's ability to provide worker groups on multiple platforms - can you confirm if that's the case, or if it is mutually exclusive? (i.e. only eks is configured with only windows or linux worker groups)?

Reply:
I tried that before, but I went into terraform cycle error.

What do you think about https://github.com/terraform-aws-modules/terraform-aws-eks/pull/1413 ? Can we close your PR ?

Reply:
Looking through how `local.workers_group_defaults` is used / referenced in other places, I think this approach might work. The update would then look like: 

```
worker_has_linux_ami = local.workers_group_defaults["platform"] == "linux"
worker_has_windows_ami = local.workers_group_defaults["platform"] == "windows"
```

This is defaulted to "linux" by https://github.com/terraform-aws-modules/terraform-aws-eks/blob/master/local.tf#L75 and would only look up 1 or the other depending on default vs a var override being provided.

I'll look to start testing this approach, call out anything i'm overlooking here. If this is the case, it might not even be necessary to have this as a local here, it might be a bit simpler to just use this condition in the `count = local.workers_group_defaults["platform"] == "windows"` here: https://github.com/terraform-aws-modules/terraform-aws-eks/blob/master/data.tf#L34

Reply:
@barryib - ok - i see the cycle errors you mentioned. Sure, happy to close this 1 in favour of #1413

Reply:
ok. I'll close this. Can you please review #1413 ?